optuna/pruners/hyperband.py

```diff
@@ -70,20 +87,43 @@ class HyperbandPruner(BasePruner):
     """
 
     def __init__(
-        self, min_resource=1, reduction_factor=3, n_brackets=4, min_early_stopping_rate_low=0
-    ):
-        # type: (int, int, int, int) -> None
-
+        self,
+        min_resource: int = 1,
+        max_resource: int = 80,
+        reduction_factor: int = 3,
+        n_brackets: Optional[int] = None,
+        min_early_stopping_rate_low: int = 0,
+    ) -> None:
         self._pruners = []  # type: List[SuccessiveHalvingPruner]
         self._reduction_factor = reduction_factor
         self._resource_budget = 0
-        self._n_brackets = n_brackets
+
+        if n_brackets is None:
+            # In the original paper http://www.jmlr.org/papers/volume18/16-558/16-558.pdf, the
+            # inputs of Hyperband are ``R``: max resource and ``\eta``: reduction factor. The
+            # number of brackets (this is referred as ``s_{max} + 1`` in the paper) is calculated
+            # by s_{max} + 1 = \floor{\log_{\eta} (R)} + 1 in Algorithm 1 of the original paper.
+            self._n_brackets = (
+                math.floor(math.log2(max_resource) / math.log2(reduction_factor)) + 1
+            )
+        else:
+            message = (
+                "The argument of ``n_brackets`` is deprecated. "
+                "The number of brackets is can be determined by ``max_resource`` and "
+                "``reduction_factor`` as "
+                "`n_brackets = floor(log(max_resource) / log(reduction_factor))`. "
+                "Please specify ``max_resource`` appropriately."
```

Comment:
I think we can also use single backticks in the log messages (c.f., https://github.com/optuna/optuna/blob/master/optuna/samplers/cmaes.py#L156-L161).
```suggestion
                "The argument of `n_brackets` is deprecated. "
                "The number of brackets is can be determined by `max_resource` and "
                "`reduction_factor` as "
                "`n_brackets = floor(log(max_resource) / log(reduction_factor))`. "
                "Please specify `max_resource` appropriately."
```